src/modules/custom.rs

```diff
@@ -207,6 +214,19 @@ fn exec_command(cmd: &str, shell: Option<&str>) -> Option<String> {
     }
 }
 
+/// If the specified shell refers to PowerShell, adds the arguments "-Command -" to the
+/// given command.
+fn handle_powershell(command: &mut Command, shell: &str) {
+    let is_powershell = shell.ends_with("pwsh.exe")
+        || shell.ends_with("powershell.exe")
+        || shell.ends_with("pwsh")
+        || shell.ends_with("powershell");
+
+    if is_powershell {
+        command.arg("-Command").arg("-");
```

Comment:
What is the `-` argument for? I can't see this [through the documentation](https://docs.microsoft.com/en-us/powershell/module/Microsoft.PowerShell.Core/About/about_PowerShell_exe?view=powershell-5.1). Testing it locally also outputs an error and help message.

We might want the `-NoProfile` argument on the other hand. That would speed up execution of the custom command as the profile loading is skipped. Testing locally a `hello world` echo command with and without that argument shows substantial delay (when profile is loaded). 

```suggestion
        command.arg("-NoProfile").arg("-Command");
```

Reply:
I am not sure about -NoProfile, since the user may have functions defined in their profile that they use in the prompt.

Omitting it _does_ slow down significantly the prompt. Not sure what the best course of action is.

As for `-`, it is actually part of the documentation, it's just subtle (`-Command - | ...`, see that first alternative `-`?). It means "read from stdin."

Reply:
Guys, and if we add a new param to add custom args? This attend all shells and allow the user customize which args he wants.

What do you think?

Reply:
@rcjeferson That's kinda possible now, and we can extend it further by splitting `shell` into shell + args (which seems to be automatic on Windows?), but we still need a sane default for the time being.

Reply:
Yes, in this case we can assume without args as default and put `-NoProfile` as suggested value to this new param (args) in documentation. That way, who need the profile config get by default and who not need, can add `-NoProfile` arg.

I agree that in some cases I would like to use my alias configured in my profile, but in mostly cases isn't necessary, so it will be possible disable with this new param.

What do you think about this?

Reply:
If we're going down that way, maybe `-NoProfile -Command -` should be the default, since I think users will more often run built-in commands. Then for "advanced" usages, the user may remove `-NoProfile` at the cost of speed of execution.

Reply:
Thanks @71 for the explanation around usage of `-`.

+1 for the `-NoProfile -Command -` as default, and making this a priority in the time being.

+1 for the possibility for overriding it. Could be done later if we can't squeeze into it right now?

Reply:
I think the best thing to do here would be to change `shell` from accepting a string to accepting a list of strings, but that means breaking backward compatibility.

Reply:
Commit updated for this, please take a look!

Reply:
Do we have some guidelines on breaking backward compatibility? cc @matchai 

If only @rcjeferson used that feature, I guess we're good to go but I'm wondering if more people uses this. 😅

Reply:
We don't quite yet.
https://github.com/starship/starship/issues/1075 would ideally give us the tools necessary to surface broken configuration, and link users to docs with necessary steps, but that's not yet in place. 😰

That said, I would assume that Powershell + custom modules is a rare combination and is worth pushing a fix for, noting the breaking change in our changelog.

Reply:
Another possibility I had in mind was to allow both a string (that would be internally converted to a vector) and a list, but that requires a new implementation of `ModuleConfig` (e.g. `ModuleConfig` for an enum `VectorOrString`, or `ModuleConfig` for a new type `VecOrString<'a>(Vec<&'a str>)` that can handle both strings and lists in TOML).

Reply:
I think that sounds like a worthwhile addition 👍